src/components/account/thresholds-settings/ThresholdsSettingsForm.tsx

```diff
@@ -0,0 +1,113 @@
+'use client';
+
+import { useMutation } from '@tanstack/react-query';
+import Link from 'next/link';
+import { useTranslations } from 'next-intl';
+import { type ReactNode, useMemo } from 'react';
+
+import type { TypeThreshold, TypeThresholdType } from '@/__generated__/data-contracts';
+import { patchUsersMe } from '@/actions/patchUsersMe';
+import Form from '@/components/form/Form';
+import { FormInput } from '@/components/form/Input';
+import Button from '@/ui/button/button';
+import ButtonGroup from '@/ui/button-group/button-group';
+import Label from '@/ui/label/label';
+import NotificationBanner from '@/ui/notification-banner/notification-banner';
+import { Table, TableBody, TableCell, TableCellHeader, TableHead, TableRow } from '@/ui/table/Table';
+
+import { schema } from './ThresholdsSettingsFormSchema';
+
+type ThresholdsSettingsFormProps = {
+  currentSettings?: TypeThreshold[] | null;
+  children: ReactNode;
+};
+
+const ThresholdsSettingsForm = ({ currentSettings, children }: ThresholdsSettingsFormProps) => {
+  const t = useTranslations('Forms.Thresholds_settings');
+  const tCommon = useTranslations('Common');
+
+  const { mutate, isSuccess, isError, isPending } = useMutation({
+    mutationFn: async (values: TypeThreshold[]) => {
+      await patchUsersMe({ notification_thresholds: values });
+    },
+  });
+
+  const defaultValues = useMemo(() => currentSettings?.reduce((acc, cur) => {
+    if (cur.type === 'TIME_TO_EVENT') {
+      acc[cur.type] = cur.value / 3600;
+    } else if (cur.type === 'PROBABILITY_OF_COLLISION') {
+      acc[cur.type] = cur.value * 100;
+    } else {
+      acc[cur.type] = cur.value;
+    }
+    return acc;
+  }, {} as Record<TypeThresholdType, number>), [currentSettings]);
+
+  const handleFormSubmit = async (values: Record<TypeThresholdType, number>) => {
+    const notificationThresholds: TypeThreshold[] = [];
+    for (const [key, fieldValue] of Object.entries(values)) {
+      let value = 0;
+      if (key === 'TIME_TO_EVENT') {
+        value = fieldValue * 3600;
+      } else if (key === 'PROBABILITY_OF_COLLISION') {
+        value = fieldValue / 100;
+      } else {
+        value = fieldValue;
+      }
+      notificationThresholds.push({
+        type: key as TypeThresholdType,
+        value,
+      });
+    }
+    await mutate(notificationThresholds);
```

Comment:
```suggestion
   mutate(notificationThresholds);
```

'await' not needed here